<html>

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 14 (filtered)">
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
@font-face
	{font-family:UICTFontTextStyleBody;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:10.0pt;
	margin-left:0in;
	line-height:115%;
	font-size:11.0pt;
	font-family:"Calibri","sans-serif";}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;}
span.apple-converted-space
	{mso-style-name:apple-converted-space;}
.MsoChpDefault
	{font-family:"Calibri","sans-serif";}
.MsoPapDefault
	{margin-bottom:10.0pt;
	line-height:115%;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.0in 1.0in 1.0in;}
div.WordSection1
	{page:WordSection1;}
-->
</style>

</head>

<body lang=EN-US link=blue vlink=purple>

<div class=WordSection1>

<p class=MsoNormalCxSpFirst style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>References</span></b></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif";
color:black'>Yaari, G. and Uduman, M. and Kleinstein, S. H. (2012). Quantifying
selection in high-throughput Immunoglobulin sequencing data sets. In<span
class=apple-converted-space>&nbsp;</span><em>Nucleic Acids Research, 40 (17),
pp. e134–e134.</em><span class=apple-converted-space><i>&nbsp;</i></span>[</span><span
lang=EN-GB><a href="http://dx.doi.org/10.1093/nar/gks457" target="_blank"><span
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif";
color:#303030'>doi:10.1093/nar/gks457</span></a></span><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif";
color:black'>][</span><span lang=EN-GB><a
href="http://dx.doi.org/10.1093/nar/gks457" target="_blank"><span
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif";
color:#303030'>Link</span></a></span><span lang=EN-GB style='font-size:12.0pt;
line-height:115%;font-family:"Times New Roman","serif";color:black'>]</span></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>Graphs</span></b></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>AA
mutation frequency</span></u></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>For
each class, the frequency of replacement mutations at each amino acid position
is shown, which is calculated by dividing the number of replacement mutations
at a particular amino acid position/the number sequences that have an amino
acid at that particular position. Since the length of the CDR1 and CDR2 region
is not the same for every VH gene, some amino acids positions are absent.
Therefore we calculate the frequency using the number of amino acids present at
that that particular location. </span></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>Antigen
selection (BASELINe)</span></u></p>

<p class=MsoNormalCxSpMiddle style='text-align:justify'><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>Shows
the results of the analysis of antigen selection as performed using BASELINe.
Details on the analysis performed by BASELINe can be found in Yaari et al,
PMID: 22641856. The settings used for the analysis are</span><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>:
focused, SHM targeting model: human Tri-nucleotide, custom bounderies. The
custom boundries are dependent on the ‘sequence starts at filter’. </span></p>

<p class=MsoNormalCxSpMiddle style='line-height:normal'><span lang=NL
style='font-family:UICTFontTextStyleBody;color:black'>Leader:
1:26:38:55:65:104:-</span></p>

<p class=MsoNormalCxSpMiddle style='line-height:normal'><span lang=NL
style='font-family:UICTFontTextStyleBody;color:black'>FR1: 27:27:38:55:65:104:-</span></p>

<p class=MsoNormalCxSpMiddle style='line-height:normal'><span lang=NL
style='font-family:UICTFontTextStyleBody;color:black'>CDR1:&nbsp;27:27:38:55:65:104:-</span></p>

<p class=MsoNormalCxSpLast style='line-height:normal'><span lang=NL
style='font-family:UICTFontTextStyleBody;color:black'>FR2:&nbsp;27:27:38:55:65:104:-</span></p>

<p class=MsoNormal><span lang=NL style='font-size:12.0pt;line-height:115%;
font-family:"Times New Roman","serif"'>Hanna IJspeert, Pauline A. van
Schouwenburg, David van Zessen, Ingrid Pico-Knijnenburg, Gertjan J. Driessen,
Andrew P. Stubbs, and Mirjam van der Burg (2016). </span><span
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>Evaluation
of the Antigen-Experienced B-Cell Receptor Repertoire in Healthy Children and
Adults. In <i>Frontiers in Immunolog, 7, pp. e410-410. </i>[<a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5066086/"><span
style='color:windowtext'>doi:10.3389/fimmu.2016.00410</span></a>][<a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5066086/"><span
style='color:windowtext'>Link</span></a>]</span></p>

</div>

</body>

</html>
