<html>

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 14 (filtered)">
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:10.0pt;
	margin-left:0in;
	line-height:115%;
	font-size:11.0pt;
	font-family:"Calibri","sans-serif";}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;}
p
	{margin-right:0in;
	margin-left:0in;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{mso-style-link:"Balloon Text Char";
	margin:0in;
	margin-bottom:.0001pt;
	font-size:8.0pt;
	font-family:"Tahoma","sans-serif";}
p.msochpdefault, li.msochpdefault, div.msochpdefault
	{mso-style-name:msochpdefault;
	margin-right:0in;
	margin-left:0in;
	font-size:12.0pt;
	font-family:"Calibri","sans-serif";}
p.msopapdefault, li.msopapdefault, div.msopapdefault
	{mso-style-name:msopapdefault;
	margin-right:0in;
	margin-bottom:10.0pt;
	margin-left:0in;
	line-height:115%;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";}
span.apple-converted-space
	{mso-style-name:apple-converted-space;}
span.BalloonTextChar
	{mso-style-name:"Balloon Text Char";
	mso-style-link:"Balloon Text";
	font-family:"Tahoma","sans-serif";}
.MsoChpDefault
	{font-size:10.0pt;
	font-family:"Calibri","sans-serif";}
.MsoPapDefault
	{margin-bottom:10.0pt;
	line-height:115%;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.0in 1.0in 1.0in;}
div.WordSection1
	{page:WordSection1;}
-->
</style>

</head>

<body lang=EN-US link=blue vlink=purple>

<div class=WordSection1>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><b><span lang=EN-GB style='color:black'>References</span></b></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span lang=EN-GB style='color:black'>Gupta,
Namita T. and Vander Heiden, Jason A. and Uduman, Mohamed and Gadala-Maria,
Daniel and Yaari, Gur and Kleinstein, Steven H. (2015). <a name="OLE_LINK106"></a><a
name="OLE_LINK107"></a>Change-O: a toolkit for analyzing large-scale B cell
immunoglobulin repertoire sequencing data: Table 1. In<span
class=apple-converted-space>&nbsp;</span><em>Bioinformatics, 31 (20), pp.
3356â€“3358.</em><span class=apple-converted-space><i>&nbsp;</i></span>[</span><a
href="http://dx.doi.org/10.1093/bioinformatics/btv359" target="_blank"><span
lang=EN-GB style='color:#303030'>doi:10.1093/bioinformatics/btv359</span></a><span
lang=EN-GB style='color:black'>][</span><a
href="http://dx.doi.org/10.1093/bioinformatics/btv359" target="_blank"><span
lang=EN-GB style='color:#303030'>Link</span></a><span lang=EN-GB
style='color:black'>]</span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span lang=EN-GB style='color:black'>&nbsp;</span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><a name="OLE_LINK110"><u><span lang=EN-GB
style='color:black'>All, IGA, IGG, IGM and IGE tabs</span></u></a></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span lang=EN-GB style='color:black'>In
these tabs information on the clonal relation of transcripts can be found. To
calculate clonal relation Change-O is used (Gupta et al, PMID: 26069265).
Transcripts are considered clonally related if they have maximal three nucleotides
difference in their CDR3 sequence and the same first V segment (as assigned by
IMGT). Results are represented in a table format showing the clone size and the
number of clones or sequences with this clone size. Change-O settings used are
the </span><span lang=EN-GB>nucleotide hamming distance substitution model with
a complete distance of maximal three. For clonal assignment the first gene
segments were used, and the distances were not normalized. In case of
asymmetric distances, the minimal distance was used.<span style='color:black'> </span></span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span lang=EN-GB style='color:black'>&nbsp;</span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><u><span lang=EN-GB style='color:black'>Overlap
tab</span></u><span lang=EN-GB style='color:black'> </span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span lang=EN-GB style='color:black'>This
tab gives information on with which (sub)classe(s) each unique analyzed region
(based on the exact nucleotide sequence of the analyzes region and the CDR3
nucleotide sequence) is found with. This gives information if the combination
of the exact same nucleotide sequence of the analyzed region and the CDR3
sequence can be found in multiple (sub)classes.</span></p>

<p style='margin-top:0in;margin-right:0in;margin-bottom:6.4pt;margin-left:0in;
text-align:justify;background:white'><span style='color:black'><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA8AAAAPCAYAAAA71pVKAAAAzElEQVQoka2TwQ2CQBBFpwTshw4ImW8ogJMlUIMmhNCDxgasAi50oSXA8XlAjCG7aqKTzGX/vsnM31mzR0gk7tTudO5MEizpzvQ4ryUSe408J3Xn+grE0p1rnpOamVmWsZG4rS+dzzAMsN8Hi9yyjI1JNGtxu4VxBJgLRLpoTKIPiW0LlwtUVRTubW2OBGUJu92cZRmdfbKQMAw8o+vi5v0fLorZ7Y9waGYJjsf38DJz0O1PsEQffOcv4Sa6YYfDDJ5Obzbsp93+5VfdATueO1fdLdI0AAAAAElFTkSuQmCC"> Please note that this tab is based on all
sequences before filter unique sequences and the remove duplicates based on
filters are applied. In this table only sequences occuring more than once are
included. </span></p>

</div>

</body>

</html>
