<html>

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 14 (filtered)">
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:10.0pt;
	margin-left:0in;
	line-height:115%;
	font-size:11.0pt;
	font-family:"Calibri","sans-serif";}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;}
p.MsoNoSpacing, li.MsoNoSpacing, div.MsoNoSpacing
	{margin:0in;
	margin-bottom:.0001pt;
	font-size:11.0pt;
	font-family:"Calibri","sans-serif";}
.MsoChpDefault
	{font-family:"Calibri","sans-serif";}
.MsoPapDefault
	{margin-bottom:10.0pt;
	line-height:115%;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.0in 1.0in 1.0in;}
div.WordSection1
	{page:WordSection1;}
-->
</style>

</head>

<body lang=EN-US link=blue vlink=purple>

<div class=WordSection1>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Info</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The complete
dataset:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Allows downloading of the complete parsed data set.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The filtered
dataset:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Allows downloading of all parsed IMGT information of all transcripts that
passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The alignment
info on the unmatched sequences:</span></u><span lang=EN-GB style='font-size:
12.0pt;font-family:"Times New Roman","serif"'> Provides information of the subclass
alignment of all unmatched sequences. For each sequence the chunck hit
percentage and the nt hit percentage is shown together with the best matched
subclass.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>SHM Overview</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The SHM Overview
table as a dataset:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Allows downloading of the SHM Overview
table as a data set.  </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Motif data per
sequence ID:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'> Provides a file that contains information for each
transcript on the number of mutations present in WA/TW and RGYW/WRCY motives.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Mutation data
per sequence ID: </span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'>Provides a file containing information
on the number of sequences bases, the number and location of mutations and the
type of mutations found in each transcript. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Base count for
every sequence:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'> links to a page showing for each transcript the
sequence of the analysed region (as dependent on the sequence starts at filter),
the assigned subclass and the number of sequenced A,C,G and T’s.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the percentage of mutations in AID and pol eta motives plot:</span></u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Provides a file containing the values used to generate the percentage of
mutations in AID and pol eta motives plot in the SHM overview tab.</span></p>

<p class=MsoNormalCxSpFirst style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>The
data used to generate the relative mutation patterns plot:</span></u><span
lang=EN-GB style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>
Provides a download with the data used to generate the relative mutation
patterns plot in the SHM overview tab.</span></p>

<p class=MsoNormalCxSpLast style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>The
data used to generate the absolute mutation patterns plot:</span></u><span
lang=EN-GB style='font-size:12.0pt;line-height:115%;font-family:"Times New Roman","serif"'>
Provides a download with the data used to generate the absolute mutation
patterns plot in the SHM overview tab. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>SHM Frequency</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data
generate the frequency scatter plot:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Allows
downloading the data used to generate the frequency scatter plot in the SHM
frequency tab. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the frequency by class plot:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Allows
downloading the data used to generate frequency by class plot included in the
SHM frequency tab.           </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for
frequency by subclass:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Provides information of the number and
percentage of sequences that have 0%, 0-2%, 2-5%, 5-10%, 10-15%, 15-20%,
&gt;20% SHM. Information is provided for each subclass.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> </span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Transition
Tables</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'all' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGA' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGA sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGA1' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGA1 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGA2' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGA2 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGG' transition plot :</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGG sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGG1' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGG1 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGG2' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGG2 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGG3' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGG3 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGG4' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGG4 sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGM' transition plot :</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the information used to
generate the transition table for all IGM sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
'IGE' transition plot:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Contains the
information used to generate the transition table for all IGE sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Antigen
selection</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>AA mutation data
per sequence ID:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'> Provides for each transcript information on whether
there is replacement mutation at each amino acid location (as defined by IMGT).
For all amino acids outside of the analysed region the value 0 is given.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Presence of AA
per sequence ID:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'> Provides for each transcript information on which
amino acid location (as defined by IMGT) is present. </span><span lang=NL
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>0 is absent, 1
is present. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the aa mutation frequency plot:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Provides the
data used to generate the aa mutation frequency plot for all sequences in the
antigen selection tab.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the aa mutation frequency plot for IGA:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>  Provides the
data used to generate the aa mutation frequency plot for all IGA sequences in
the antigen selection tab.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the aa mutation frequency plot for IGG:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Provides the
data used to generate the aa mutation frequency plot for all IGG sequences in
the antigen selection tab.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the aa mutation frequency plot for IGM:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Provides the
data used to generate the aa mutation frequency plot for all IGM sequences in
the antigen selection tab.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data used to
generate the aa mutation frequency plot for IGE:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>   Provides the
data used to generate the aa mutation frequency plot for all IGE sequences in
the antigen selection tab.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline PDF (</span></u><span
lang=EN-GB><a href="http://selection.med.yale.edu/baseline/"><span
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>http://selection.med.yale.edu/baseline/</span></a></span><u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>):</span></u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'> PDF
containing the </span><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'>Antigen selection (BASELINe) graph for all
sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline data:</span></u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Table output of the BASELINe analysis. Calculation of antigen selection as
performed by BASELINe are shown for each individual sequence and the sum of all
sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGA
PDF:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
PDF containing the </span><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'>Antigen selection (BASELINe) graph for all
sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGA
data:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Table output of the BASELINe analysis. Calculation of antigen selection as
performed by BASELINe are shown for each individual IGA sequence and the sum of
all IGA sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGG
PDF:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
PDF containing the </span><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'>Antigen selection (BASELINe) graph for all IGG
sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGG
data:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Table output of the BASELINe analysis. Calculation of antigen selection as
performed by BASELINe are shown for each individual IGG sequence and the sum of
all IGG sequences.        </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGM PDF:</span></u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'> PDF
containing the </span><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'>Antigen selection (BASELINe) graph for all IGM
sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGM
data:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Table output of the BASELINe analysis. Calculation of antigen selection as
performed by BASELINe are shown for each individual IGM sequence and the sum of
all IGM sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGE
PDF:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
PDF containing the </span><span lang=EN-GB style='font-size:12.0pt;font-family:
"Times New Roman","serif"'>Antigen selection (BASELINe) graph for all IGE
sequences.</span><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Baseline IGE
data:</span></u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Table output of the BASELINe analysis. Calculation of antigen selection as
performed by BASELINe are shown for each individual IGE sequence and the sum of
all IGE sequences.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>CSR</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
</span></u><u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>IGA
subclass distribution plot :</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> </span><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Data used for
the generation of the </span><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'>IGA subclass distribution plot provided
in the CSR tab. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The data for the
</span></u><u><span lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>IGA
subclass distribution plot :</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Data used for the generation of the </span><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>IGG
subclass distribution plot provided in the CSR tab. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=NL
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Clonal relation</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Sequence overlap
between subclasses:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Link to the overlap table as provided
under the clonality overlap tab.         </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
file with defined clones and subclass annotation:</span></u><span
lang=EN-GB style='font-size:12.0pt;font-family:"Times New Roman","serif"'>
Downloads a table with the calculation of clonal relation between all
sequences. For each individual transcript the results of the clonal assignment
as provided by Change-O are provided. Sequences with the same number in the CLONE
column are considered clonally related. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
defined clones summary file:</span></u><span lang=EN-GB style='font-size:12.0pt;
font-family:"Times New Roman","serif"'> Gives a summary of the total number of
clones in all sequences and their clone size.           </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
file with defined clones of IGA:</span></u><span lang=EN-GB style='font-size:
12.0pt;font-family:"Times New Roman","serif"'> Downloads a table with the
calculation of clonal relation between all IGA sequences. For each individual
transcript the results of the clonal assignment as provided by Change-O are
provided. Sequences with the same number in the CLONE column are considered
clonally related. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
defined clones summary file of IGA:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Gives a summary
of the total number of clones in all IGA sequences and their clone size.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
file with defined clones of IGG:</span></u><span lang=EN-GB style='font-size:
12.0pt;font-family:"Times New Roman","serif"'> Downloads a table with the
calculation of clonal relation between all IGG sequences. For each individual
transcript the results of the clonal assignment as provided by Change-O are
provided. Sequences with the same number in the CLONE column are considered
clonally related. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
defined clones summary file of IGG:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Gives a summary
of the total number of clones in all IGG sequences and their clone size.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
file with defined clones of IGM:</span></u><span lang=EN-GB style='font-size:
12.0pt;font-family:"Times New Roman","serif"'> Downloads a table
with the calculation of clonal relation between all IGM sequences. For each
individual transcript the results of the clonal assignment as provided by
Change-O are provided. Sequences with the same number in the CLONE column are
considered clonally related. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
defined clones summary file of IGM:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Gives a summary
of the total number of clones in all IGM sequences and their clone size.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
file with defined clones of IGE:</span></u><span lang=EN-GB style='font-size:
12.0pt;font-family:"Times New Roman","serif"'> Downloads a table with the
calculation of clonal relation between all IGE sequences. For each individual
transcript the results of the clonal assignment as provided by Change-O are
provided. Sequences with the same number in the CLONE column are considered
clonally related. </span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>The Change-O DB
defined clones summary file of IGE:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Gives a summary
of the total number of clones in all IGE sequences and their clone size.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><b><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>Filtered IMGT
output files</span></b></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGA sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGA
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGA1 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGA1
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGA2 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a .txz
file with the same format as downloaded IMGT files that contains all IGA2
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGG sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a .txz
file with the same format as downloaded IMGT files that contains all IGG
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGG1 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGG1
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGG2 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGG2
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGG3 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a .txz
file with the same format as downloaded IMGT files that contains all IGG3
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGG4 sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGG4
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGM sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a .txz
file with the same format as downloaded IMGT files that contains all IGM
sequences that have passed the chosen filter settings.</span></p>

<p class=MsoNoSpacing style='text-align:justify'><u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'>An IMGT archive
with just the matched and filtered IGE sequences:</span></u><span lang=EN-GB
style='font-size:12.0pt;font-family:"Times New Roman","serif"'> Downloads a
.txz file with the same format as downloaded IMGT files that contains all IGE
sequences that have passed the chosen filter settings.</span></p>

</div>

</body>

</html>
